openmetadata-service/src/main/java/org/openmetadata/service/resources/teams/UserResource.java

```diff
@@ -1131,6 +1205,112 @@ public Response loginUserWithPassword(
     }
   }
 
+  private Response handleLdapAuthLogin(UriInfo uriInfo, LoginRequest req) {
```

Comment:
can we move all of this code out of Resource into a separate class of its own, probably AuthenticationHandlers and possibly an interface to extend from. We have username/password , SAML and Ldap. All of these existing in its own method doesn't look right to me.
We can build a interface to implement the different mechanisms